lib/assessment.js

```diff
@@ -29,7 +47,10 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  for (var i = 0; i < length; i++) {
+    array[i] = defaultsFunction(i);
```

Comment:
What are you assigning to? You don't have an empty array stored in a variable called `array`.